app/gui/src/main/scala/org/bitcoins/gui/ContractGUI.scala

```diff
@@ -0,0 +1,274 @@
+    styleClass += "small"
+    children = Seq(fileChooserButton, addContractTF)
+  }
+
+  private lazy val eventIdCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Event Id"
+    prefWidth = 160
+    cellValueFactory = { status =>
+      val eventIdStr = status.value._1.eventTLV.eventId
+      new StringProperty(status, "Event Id", eventIdStr)
+    }
+  }
+
+  private val ISO_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'")
+
+  private lazy val maturityCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Matures"
+    prefWidth = 130
+    cellValueFactory = { status =>
+      val d = new Date(
+        status.value._1.eventTLV.eventMaturityEpoch.toLong * 1000)
+      new StringProperty(status, "Matures", ISO_FORMAT.format(d))
+    }
+  }
+
+  private lazy val contractCol = new TableColumn[
+    (OracleAnnouncementV0TLV, Option[ContractInfoV0TLV]),
+    String] {
+    text = "Contract"
+    prefWidth = 160
+    cellValueFactory = { status =>
+      val contractStr =
+        status.value._2 match {
+          case None        => ""
+          case Some(value) => value.contractDescriptor.toString()
+        }
+      new StringProperty(status, "Contract", contractStr)
+    }
+  }
+
+  private lazy val eventTableView =
+    new TableView[(OracleAnnouncementV0TLV, Option[ContractInfoV0TLV])](
+      GlobalContractData.announcements) {
+      columns ++= Seq(eventIdCol, maturityCol, contractCol)
+      sortOrder.add(maturityCol)
+
+      val removeContract: MenuItem = new MenuItem("Remove Contract") {
+        onAction = _ => {
+          val selected = selectionModel().getSelectedItem
+          model.onRemoveContract(selected._1, selected._2)
+        }
+      }
+
+      contextMenu = new ContextMenu() {
+        items += removeContract
+      }
+
+      onMouseClicked = _ => {
+        val i = selectionModel().getSelectedItem
+        if (i != null) {
+          showCreateOfferPane(i._1, i._2)
+        }
+      }
+    }
+
+  lazy val eventPane = new VBox {
+    padding = Insets(0)
+    children = Seq(eventTableView)
+  }
+
+  // Text paste / file browse handler
+  private def onContractAdded(hex: String, file: Option[File]): Boolean = {
+    var success = true
+    // Accept / Sign / Broadcast identifier
+    LnMessageFactory(DLCOfferTLV).fromHexT(hex) match { // Accept
+      case Success(_) =>
+        showAcceptOfferPane(hex)
+      case Failure(_) =>
+        LnMessageFactory(DLCAcceptTLV).fromHexT(hex) match { // Sign
+          case Success(_) =>
+            showSignDLCPane(hex, file)
+          case Failure(_) =>
+            LnMessageFactory(DLCSignTLV).fromHexT(hex) match { // Broadcast
+              case Success(_) =>
+                showBroadcastDLCPane(hex, file)
+              case Failure(_) => success = false // Nothing else to check for
+            }
+        }
+    }
+    success
+  }
```

Comment:
instead of using a `var` here you can just make every case return a `true` and the final failure return a `false`